Keda.CosmosDbScaler.Demo.OrderProcessor/Worker.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Keda.CosmosDbScaler.Demo.Shared;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Keda.CosmosDbScaler.Demo.OrderProcessor
+{
+    internal sealed class Worker : BackgroundService
+    {
+        private readonly CosmosDbConfig _cosmosDbConfig;
+        private readonly ILogger<Worker> _logger;
+
+        private ChangeFeedProcessor _processor;
+
+        public Worker(CosmosDbConfig cosmosDbConfig, ILogger<Worker> logger)
+        {
+            _cosmosDbConfig = cosmosDbConfig ?? throw new ArgumentNullException(nameof(cosmosDbConfig));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public override async Task StartAsync(CancellationToken stoppingToken)
+        {
+            Database leaseDatabase = await new CosmosClient(_cosmosDbConfig.LeaseConnection)
+                .CreateDatabaseIfNotExistsAsync(_cosmosDbConfig.LeaseDatabaseId);
+
+            Container leaseContainer = await leaseDatabase
+                .CreateContainerIfNotExistsAsync(
+                    new ContainerProperties(_cosmosDbConfig.LeaseContainerId, partitionKeyPath: "/id"),
+                    throughput: 400);
+
+            // Change feed processor instance name should be unique for each container application.
+            string instanceName = $"Instance-{Dns.GetHostName()}";
+
+            _processor = new CosmosClient(_cosmosDbConfig.Connection)
```

Comment:
if the Connection of the lease is the same (LeaseConnection == Connection) you don't need 2 clients

Reply:
Since this one is a demo application, I thought to not add an extra if-check for conciseness and readability. But, since the order-processor application might stay running for long time and would keep referencing the `CosmosClient` instances during the lifetime, I will get it done. BTW, the scaler application has `CosmosDbFactory` class that ensures that the `CosmosClient` instance is re-used.

Reply:
Skipping this one since it would complicate the implementation. The code run in a sample application; it won't be packaged, deployed or be part of customer's production services.